README.rst

```diff
@@ -17,3 +17,129 @@
 
 CI Infrastructure for Apache Airflow
 ====================================
+
+Apache Airflow uses Github Actions as its Continuous Integration (CI) tool. When Airflow contributors open a pull
+request, Github Actions runs a series of tests on Apache organization runners. However, due to the limited resources
+of these runners, the tests could take a significant amount of time, sometimes up to 2 hours. To improve efficiency
+for active committers, a separate set of runners is available for them. These CI runs are executed on self-hosted
+runners deployed on an Amazon Elastic Kubernetes Service (EKS) cluster.
```

Comment:
I think that it's important to mention what operates the runs. Also, "ARC" is later used as an acronym.

```suggestion
runners deployed on an Amazon Elastic Kubernetes Service (EKS) cluster, using [Actions Runner Controller](https://github.com/actions/actions-runner-controller) (ARC).
```